### PR TITLE
Add help menu

### PR DESCRIPTION
**Pull request by zunda-arrow**
Adds a simple help menu. I haven't bothered with a delete button yet.

### CODE REVIEW

File: bot/constants.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import hikari
+
+EMBED_COLOR = hikari.Color.from_hex_code("#F7B159")
```
Comment (circuitsacul) [inline bot/constants.py]: Why not `0xF7B159`? That's a valid value to pass to the embed builder anyways

File: bot/plugins/help.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+plugin = Plugin()
+
+HELP_EMBEDS = [
+    hikari.Embed(
+        description=(
+            "Hi! My name is io, and my job is to run code."
+            "\n I can run any code inside of code blocks:"
+            "\n\\`\\`\\`<language-name>"
+            "\n<your code here>"
+            "\n\\`\\`\\`"
```
Comment (circuitsacul) [inline bot/plugins/help.py]: You can use zero-width spaces between the back-ticks instead. Then surround the whole thing in a code block. The result is this:

Comment (circuitsacul) [reply bot/plugins/help.py]: If you think the non-code-block version is more readable I'm willing to change my mind. Just thought I'd mention that this *is* possible too.

File: bot/plugins/help.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        ),
+        color=EMBED_COLOR,
+    ),
+    hikari.Embed(
+        description=(
+            "\n\\* Running code - Start your message with `io/run`."
+            "\n\\* View Assembly - Start your message with `io/asm`."
```
Comment (circuitsacul) [inline bot/plugins/help.py]: Why not use discord's list markdown? Start each line with `-` (no space before)

File: bot/plugins/help.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            "command."
+        ),
+        color=EMBED_COLOR,
+    ),
+]
+
+
+@plugin.include
+@crescent.command
+async def help(ctx: crescent.Context):
```
Comment (circuitsacul) [inline bot/plugins/help.py]: -> None

File: bot/plugins/help.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    await ctx.respond(embeds=HELP_EMBEDS)
+
+
+@plugin.include
+@crescent.event
+async def on_message(message: hikari.MessageCreateEvent):
```
Comment (circuitsacul) [inline bot/plugins/help.py]: -> None

File: bot/plugins/help.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    me = plugin.app.get_me()
+
+    if not me:
+        return
```
Comment (circuitsacul) [inline bot/plugins/help.py:47]: Maybe this? Up to you though
```py
if (not me := plugin.app.get_me()):
    return
```

File: bot/plugins/help.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    if message.author_id == me.id:
+        return
+
+    if not (
+        message.content == me.mention
+        or message.content.removeprefix(me.mention).strip() == "help"
+    ):
+        return
+
+    await message.message.respond(embeds=HELP_EMBEDS)
```
Comment (circuitsacul) [inline bot/plugins/help.py]: Doesn't this mean that saying "help" will trigger the bot? I think that commands besides io/run and io/asm should be slash commands. So @mention the bot (with no content), or use /help.

**Review by circuitsacul — CHANGES_REQUESTED** [review]

**Comment by circuitsacul** [comment]
I need to setup workflows lol

File: bot/plugins/help.py (hunk 2 above)
Comment (zunda-arrow) [reply bot/plugins/help.py]: i think the code block looks better

File: bot/plugins/help.py (hunk 6 above)
Comment (zunda-arrow) [reply bot/plugins/help.py:47]: i think its more readable with more if statements

File: bot/plugins/help.py (hunk 7 above)
Comment (zunda-arrow) [reply bot/plugins/help.py]: fixed

File: bot/plugins/help.py (hunk 3 above)
Comment (zunda-arrow) [reply bot/plugins/help.py]: i switched to discord's markdown list

File: bot/plugins/help.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+@plugin.include
+@crescent.event
+async def on_message(message: hikari.MessageCreateEvent) -> None:
+    me = plugin.app.get_me()
+
+    if not me:
+        return
+
+    if message.author_id == me.id:
+        return
```
Comment (circuitsacul) [inline bot/plugins/help.py]: Instead of this, just use `if not message.is_human` I think

File: bot/plugins/help.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+    if not (
+        message.content == me.mention
+        or (
+            message.content
+            and message.content.startswith(me.mention)
+            and message.content.removeprefix(me.mention).strip() == "help"
+        )
+    ):
```
Comment (circuitsacul) [inline bot/plugins/help.py]: Do we also want to add `message.content == "io/help"`?

File: bot/plugins/help.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        return
+
+    await message.message.respond(embeds=HELP_EMBEDS)
```
Comment (circuitsacul) [inline bot/plugins/help.py]: do we want reply=True, so that it actually replies to the message?

**Review by circuitsacul — CHANGES_REQUESTED** [review]

File: bot/plugins/help.py (hunk 9 above)
Comment (zunda-arrow) [reply bot/plugins/help.py]: i think so

File: bot/plugins/help.py (hunk 10 above)
Comment (zunda-arrow) [reply bot/plugins/help.py]: yeah

File: bot/plugins/help.py (hunk 9 above)
Comment (zunda-arrow) [reply bot/plugins/help.py]: it turns out supporting this requires some moderate changes to the code if i don't want to make it really bad

File: bot/plugins/help.py (hunk 8 above)
Comment (zunda-arrow) [reply bot/plugins/help.py]: done, totally forgot about that lol

**Review by circuitsacul — APPROVED** [review]

**Pull request merged**